### PR TITLE
Added table footer

### PR DESCRIPTION
**Pull request by prajapati-parth**
I think we should add footer feature since it has come across may times in issues and is quite essential. Please review.

### CODE REVIEW

File: examples/js/footer/footer-table.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+addProducts(70);
+
+export default class FooterTable extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const options = {
+      footerData: [
```
Comment (prajapati-parth) [inline examples/js/footer/footer-table.js]: @AllenFang You might want to take a look here, if passing footerData in `options` is a preferred way or it should be passed as a prop to `BootstrapTable`. Please suggest.

File: examples/js/footer/footer-table.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                label += tableData[i].price;
+              }
+              return (
+                <strong>{ label }</strong>
+              );
+            }
+          }
+        ]
+      ],
+      footerFormatterReturnData: products
```
Comment (prajapati-parth) [inline examples/js/footer/footer-table.js]: `footerFormatterReturnData` value passed here is returned as `tableData` parameter in `formatter`(line 39 above :point_up_2:). Well I'm also open to any other way if you suggest.
Thanks.

File: src/BootstrapTable.js [hunk 3]
```diff
@@ -344,6 +345,9 @@ class BootstrapTable extends Component {
     this._adjustTable();
     window.addEventListener('resize', this._adjustTable);
     this.refs.body.refs.container.addEventListener('scroll', this._scrollHeader);
+    if (this.props.showFooter) {
```
Comment (prajapati-parth) [inline src/BootstrapTable.js]: @AllenFang This check, adds event listener only if `showFooter` is `true`. This avoids reference errors when footer is not required. Same has been done on line 360 :point_down:.

**Comment by AllenFang** [comment]
@prajapati-parth thanks your contribution, I'll review it ASAP, thanks again 👍

File: examples/js/footer/footer-table.js (hunk 1 above)
Comment (AllenFang) [reply examples/js/footer/footer-table.js]: @prajapati-parth sorry for lately reply....  I'll prefer on `BootstrapTable`

**Comment by AllenFang** [comment]
I'll start review this PR from this weekend, hope we can release it in near future, thanks 👍

File: examples/js/footer/footer-table.js [hunk 4]
```diff
@@ -25,36 +24,34 @@ export default class FooterTable extends React.Component {
...
+        }
+      ]
+    ];
+
     return (
       <div>
         <BootstrapTable
           data={ products }
-          options={ options }
+          footerData={ footerData }
```
Comment (prajapati-parth) [inline examples/js/footer/footer-table.js]: @AllenFang I've removed `footerData` from `options` and added as a prop to `BootstrapTable`.

File: examples/js/footer/footer-table.js [hunk 5]
```diff
@@ -25,36 +24,34 @@ export default class FooterTable extends React.Component {
...
+            for (let i = 0, tableDataLen = tableData.length; i < tableDataLen; i++) {
+              label += tableData[i].price;
             }
+            return (
+              <strong>{ label }</strong>
+            );
           }
-        ]
-      ],
-      footerFormatterReturnData: products
```
Comment (prajapati-parth) [inline examples/js/footer/footer-table.js]: @AllenFang 
Removed `footerFormatterReturnData` from `options`.

File: src/BootstrapTable.js [hunk 6]
```diff
@@ -421,7 +420,7 @@ class BootstrapTable extends Component {
       expandColumnOptions.expandColumnBeforeSelectColumn = true;
     }
     const colGroups = Util.renderColGroup(columns, this.props.selectRow, expandColumnOptions, this.props.version);
-    const tableFooter = this.renderTableFooter(footerData, footerFormatterReturnData, columns, colGroups);
+    const tableFooter = this.renderTableFooter(this.props.footerData, this.state.data, columns, colGroups);
```
Comment (prajapati-parth) [inline src/BootstrapTable.js]: @AllenFang 
As I removed `footerFormatterData` from `options`, I'm using `this.state.data` to send back data to footerdata formatter.

**Comment by AllenFang** [comment]
@prajapati-parth I'll merge your code into `master` but will not announce this feature, because I think most of table footer need the `colspan`, but I need some time to implement this in current design.

Anyway, thanks your hard work.

**Comment by prajapati-parth** [comment]
@AllenFang ok! I'll try to put in `colspan`. We can keep this PR open.

**Comment by AllenFang** [comment]
@prajapati-parth that's ok, I'll merge it and I also fix some couple bug, I'll let you know the details later.

After this released, I will create a branch for enhancement table footer :)

**Comment by AllenFang** [comment]
I merge it manually, and your code already go into `master` branch and create `enhance-table-footer` branch, you can open PR to this branch but remember dont commit any changes under `dist` folder, thanks

**Comment by prajapati-parth** [comment]
Sure Allen!

**Comment by Skeetfly** [comment]
@prajapati-parth Can you please let me know when footer released?

**Comment by prajapati-parth** [comment]
@Skeetfly Footer feature has been released but we did not announce the feature as it needs some enhancement. 

You can refer https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/footer/footer-table.js for usage example.

Thanks.

**Pull request closed without merging**